File: backend/src/main/java/ru/skfu/moviecollection/control/dto/MovieDto.java
package ru.skfu.moviecollection.control.dto;

import java.util.UUID;
import ru.skfu.moviecollection.entity.WatchStatus;

public record MovieDto(
        UUID id,
        String title,
        int releaseYear,
        String director,
        String coverUrl,
        String category,
        WatchStatus status,
        Integer rating,
        boolean favorite
) {
}

File: mobile/app/src/main/java/ru/skfu/moviecollection/model/MovieDto.kt
package ru.skfu.moviecollection.model

import java.util.UUID

data class MovieDto(
    val id: UUID,
    val title: String,
    val releaseYear: Int,
    val director: String?,
    val coverUrl: String?,
    val category: String?,
    val status: WatchStatus,
    val rating: Int?,
    val favorite: Boolean
)

enum class WatchStatus(val label: String) {
    PLANNED("Планирую"),
    WATCHING("Смотрю"),
    WATCHED("Просмотрено"),
    DROPPED("Брошено")
}

